### PR TITLE
feat: Add legacy K&R2004 Runge Kutta solver

### PR DESCRIPTION
**Pull request by adigitoleo**
Adds a method that replicates the original RK solver, almost as a drop-in replacement for Mineral.update_orientations except a few extra arguments are required in the call signature.

### CODE REVIEW

**Comment by adigitoleo** [comment]
Don't merge yet, it needs a n_grains loop in there.

File: src/pydrex/legacy.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    get_velocity_gradient,
+    pathline,
+    faux_gridres,
+    velocity_magnitude,
+    n_steps,
+    **kwargs,
+):
+    """Implements the original RK45 approach (subroutine STRAIN).
+
+    Almost a drop-in replacement for `pdyrex.Mineral.update_orientations`,
```
Comment (Patol75) [inline src/pydrex/legacy.py:21]: pydrex

File: src/pydrex/legacy.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    but requires a few additional arguments:
+    - `faux_gridres`: local 'grid resolution' or an approximation of the spatial
```
Comment (Patol75) [inline src/pydrex/legacy.py:23]: What do you mean by `faux`?

**Review by Patol75 — COMMENTED** [review]
Should be good to go :)

File: src/pydrex/legacy.py (hunk 2 above)
Comment (adigitoleo) [reply src/pydrex/legacy.py]: Changed to gridstep

**Review by Patol75 — DISMISSED** [review]

**Comment by Patol75** [comment]
We are getting some `IndexError` following the changes to the test file.

**Review by Patol75 — APPROVED** [review]

**Pull request merged**